### ISSUE TITLE
mutate_ functions fail with non-standard data frame column names

### ISSUE THREAD

**Issue by lalj** [hadley/dplyr#2301]
If data frame column names have special characters (e.g. space or -), mutate_ does not work:

```
library(dplyr)
library(tidyr)

# works
df <- data.frame(key = c("a","b"), value = c(1,2))
df_spread <- df %>% spread(key, value)
mutate_if(df_spread, is.numeric, function(x) {x+1})

# fails with : Error in eval(expr, envir, enclos) : object 'b' not found
df <- data.frame(key = c("a","b-a"), value = c(1,2))
df_spread <- df %>% spread(key, value)
mutate_if(df_spread, is.numeric, function(x) {x+1})

# fails with: Error in parse(text = x) : <text>:1:3: unexpected symbol
df <- data.frame(key = c("a","c d"), value = c(1,2))
df_spread <- df %>% spread(key, value)
mutate_if(df_spread, is.numeric, function(x) {x+1})
```

**Comment by krlmlr** [comment]
Thanks, confirmed. Would you like to contribute a test case and perhaps a fix?

**Comment by markriseley** [comment]
This looks like it's covered by #2243

**Comment by krlmlr** [comment]
@markriseley: Thanks a lot! I did search for existing issues, but #2243 didn't contain the text mutate_if (until right now). Serves me right...

**Comment by markriseley** [comment]
I might have found a fix for this, see my comments in #2243
